gateway/managers/syncman/store_postgres.go

```diff
@@ -0,0 +1,473 @@
+					letsEncrypt := s.globalConfig.Projects[project].LetsEncrypt
+					if !reflect.DeepEqual(letsEncrypt, resource) {
+						evenType, resourceID, resourceType, resource := onAddOrUpdateResource(config.ResourceAddEvent, obj)
+						if resource == nil || resourceID == "" {
+							return
+						}
+						cb(evenType, resourceID, resourceType, resource)
+					}
+				case config.ResourceCluster:
+					cluster := s.globalConfig.ClusterConfig
+					if !reflect.DeepEqual(cluster, resource) {
+						evenType, resourceID, resourceType, resource := onAddOrUpdateResource(config.ResourceAddEvent, obj)
+						if resource == nil || resourceID == "" {
+							return
+						}
+						cb(evenType, resourceID, resourceType, resource)
+					}
+				case config.ResourceIntegration:
+					integrationconfig, ok := s.globalConfig.Integrations[resourceID]
+					if ok {
+						if !reflect.DeepEqual(integrationconfig, resource) {
+							evenType, resourceID, resourceType, resource := onAddOrUpdateResource(config.ResourceUpdateEvent, obj)
+							if resource == nil || resourceID == "" {
+								return
+							}
+							cb(evenType, resourceID, resourceType, resource)
+						}
+					} else {
+						evenType, resourceID, resourceType, resource := onAddOrUpdateResource(config.ResourceAddEvent, obj)
+						if resource == nil || resourceID == "" {
+							return
+						}
+						cb(evenType, resourceID, resourceType, resource)
+					}
+				case config.ResourceIntegrationHook:
+					integrationHook, ok := s.globalConfig.IntegrationHooks[resourceID]
+					if ok {
+						if !reflect.DeepEqual(integrationHook, resource) {
+							evenType, resourceID, resourceType, resource := onAddOrUpdateResource(config.ResourceUpdateEvent, obj)
+							if resource == nil || resourceID == "" {
+								return
+							}
+							cb(evenType, resourceID, resourceType, resource)
+						}
+					} else {
+						evenType, resourceID, resourceType, resource := onAddOrUpdateResource(config.ResourceAddEvent, obj)
+						if resource == nil || resourceID == "" {
+							return
+						}
+						cb(evenType, resourceID, resourceType, resource)
+					}
+				}
+			}
+			err = rows.Err()
+			if err != nil {
+				_ = helpers.Logger.LogError(helpers.GetRequestID(context.TODO()), "Error was encountered during iteration of rows in database", err, nil)
+				return
+			}
+		}
+	}()
+	return nil
+}
+
+// SetResource sets the project of the postgres globalConfig
+func (s *PostgresStore) SetResource(ctx context.Context, resourceID string, resource interface{}) error {
+	if err := updateResource(ctx, config.ResourceAddEvent, s.globalConfig, resourceID, "", resource); err != nil {
+		return err
+	}
+	clusterID, projectID, resourceType, err := splitResourceID(ctx, resourceID)
+	res, err := json.Marshal(resource)
+	if err != nil {
+		return err
+	}
+
+	query := "SELECT COUNT(*) as count FROM SC_CONFIG where resourceid=$1"
+	row := s.db.QueryRow(query, resourceID)
+	var count int
+	err = row.Scan(&count)
```

Comment:
I am not sure about this SQL statement.
You are doing an aggregation operation here, which will always give a single record at least with the value count which can be 0.
Your logic will work if you do select * statement